packages/vm/src/environment.rs

```diff
@@ -393,6 +458,7 @@ impl<S: Storage, Q: Querier> ContextData<S, Q> {
             querier: None,
             wasmer_instance: None,
             serialized_env: None,
+            dynamic_callstack: Vec::new(),
```

Comment:
Should not the caller contract address be contained in `dynamic_callstack` in the init? Where is this instantiate is done?

Reply:
please see the this line https://github.com/line/cosmwasm/blob/b83dbbc2d4956e1bfb762817fef11a597d306187/packages/vm/src/dynamic_link.rs#L50 .
it will be appended the caller address by above line. if when calling the A -> B, it records the address of A.  
it should be an empty stack since no dynamic call has occurred initially.

Reply:
I miss when the address is pushed to stack and stack is checked.
When calling the A -> B, checking B is not in the `dynamic_callstack` is needed, right? Where is it checked? (I misunderstand this is checked in `try_record_dynmic_call_trace`)

Reply:
yes, that's right :)